src/ruby/spec/pb/package_with_underscore/checker_spec.rb

```diff
@@ -0,0 +1,60 @@
+# Copyright 2016 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'open3'
+require 'tmpdir'
+
+def can_run_codegen_check
+  system('which grpc_ruby_plugin') && system('which protoc')
+end
+
+describe 'Ping protobuf code generation' do
+  if !can_run_codegen_check
```

Comment:
I'd like to remove this conditional test run before merging, since as is it looks like this will only be run manually.

I suggest editing [build_ruby.sh](https://github.com/grpc/grpc/blob/master/tools/run_tests/helper_scripts/build_ruby.sh#L25) to run something like:

```
make grpc_ruby_plugin -j8
```

.... in order to get the protoc and grpc_ruby_plugin executables available for the test.

This test can then find `protoc` and `grpc_ruby_plugin` under `bins/{opt,dbg}/protobuf/protoc` and `bins/{opt,dbg}/grpc_ruby_plugin` (the test will want to use those excutables directly.

Reply:
Is the protobuf code generation example group from https://github.com/grpc/grpc/blob/master/src/ruby/spec/pb/health/checker_spec.rb a good example to use for this spec?

And newb question, how do I ensure that my protos are built ahead of time like the `health_services_pb` service found at https://github.com/grpc/grpc/blob/master/src/ruby/spec/pb/health/checker_spec.rb#39? I'm guessing I have to do a little more work than just adding the `make` line, right?

Reply:
Basically, I have no idea how to generate the `package_with_underscore/service_pb.rb` file

Reply:
`make grpc_ruby_plugin` would just build the `protoc` and `grpc_ruby_plugin` binaries from the current state of the repo, and make them available for the test; it wouldn't generate the `_pb.rb` files, if that's what your asking.

I suggest then invoking those binaries during the test, to generate `_pb.rb` files "under test" with something along the lines of:

```
bins/{opt,dbg}/protobuf/protoc \
    --plugin=protoc-gen-grpc=bins/{opt,dbg}/grpc_ruby_plugin \
    --ruby_out=lib \
    --grpc_out=lib \
    src/proto/grpc/testing/package_with_underscore/service.proto \
    src/proto/grpc/testing/package_with_underscore/data.proto
```

Not sure if that answers your question?

Reply:
I believe it does. Hopefully I can get this done over the weekend :)

Reply:
cool, thanks for working on this :)